includes/class-sensei-settings.php

```diff
@@ -46,6 +44,9 @@ public function __construct() {
 		$this->register_hook_listener();
 		$this->get_settings();
 
+		// Flush rewrite rules on settings update.
+		add_action( 'init', array( $this, 'flush_rewrite_rules_on_update' ), 10, 0 );
```

Comment:
Maybe using the `admin_init` hooks is more appropriate? If that's the case, the `is_admin` check is not needed.